Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -36,126 +36,393 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/*! @abstract Indicates a @link WKWebExtensionContext @/link error. */
 WK_EXTERN NSErrorDomain const _WKWebExtensionContextErrorDomain WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
 
+/*!
+ @abstract Constants used by NSError to indicate errors in the @link WKWebExtensionContext @/link domain.
+ @constant WKWebExtensionContextErrorUnknown  Indicates that an unknown error occurred.
+ @constant WKWebExtensionContextErrorAlreadyLoaded  Indicates that the context is already loaded by a @link WKWebExtensionController @/link.
+ @constant WKWebExtensionContextErrorNotLoaded  Indicates that the context is not loaded by a @link WKWebExtensionController @/link.
+ @constant WKWebExtensionContextErrorBaseURLTaken  Indicates that another context is already using the specified base URL.
+ */
 typedef NS_ERROR_ENUM(_WKWebExtensionContextErrorDomain, _WKWebExtensionContextError) {
     _WKWebExtensionContextErrorUnknown = 1,
     _WKWebExtensionContextErrorAlreadyLoaded,
     _WKWebExtensionContextErrorNotLoaded,
     _WKWebExtensionContextErrorBaseURLTaken,
 } NS_SWIFT_NAME(_WKWebExtensionContext.Error) WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
 
+/*!
+ @abstract Constants used to indicate permission state in @link WKWebExtensionContext @/link.
+ @constant WKWebExtensionContextPermissionStateDeniedExplicitly  Indicates that the permission was explicitly denied.
+ @constant WKWebExtensionContextPermissionStateDeniedImplicitly  Indicates that the permission was implicitly denied because of another explicitly denied permission.
+ @constant WKWebExtensionContextPermissionStateRequestedImplicitly  Indicates that the permission was implicitly requested because of another explicitly requested permission.
+ @constant WKWebExtensionContextPermissionStateUnknown  Indicates that an unknown permission state.
+ @constant WKWebExtensionContextPermissionStateRequestedExplicitly  Indicates that the permission was explicitly requested.
+ @constant WKWebExtensionContextPermissionStateGrantedImplicitly  Indicates that the permission was implicitly granted because of another explicitly granted permission.
+ @constant WKWebExtensionContextPermissionStateGrantedExplicitly  Indicates that the permission was explicitly granted permission.
+ */
 typedef NS_ENUM(NSInteger, _WKWebExtensionContextPermissionState) {
     _WKWebExtensionContextPermissionStateDeniedExplicitly    = -3,
     _WKWebExtensionContextPermissionStateDeniedImplicitly    = -2,
     _WKWebExtensionContextPermissionStateRequestedImplicitly = -1,
-    _WKWebExtensionContextPermissionStateUnknown             = 0,
-    _WKWebExtensionContextPermissionStateRequestedExplicitly = 1,
-    _WKWebExtensionContextPermissionStateGrantedImplicitly   = 2,
-    _WKWebExtensionContextPermissionStateGrantedExplicitly   = 3,
+    _WKWebExtensionContextPermissionStateUnknown             =  0,
```

Comment:
Is this the suggested coding style?

Reply:
WebKit does usually line up the numbers in enums.